### PR TITLE
Port `transfer-circuits` to `dusk-plonk@0.13`

### LINKED ISSUE #764: Port the transfer circuits to `plonk@0.13`
#### Summary
The circuits should be ported to use `plonk` version `0.13`. The API has changed significantly and numerous improvement in terms of speed and memory efficiency have been made. It is better if our networks can make use of them.

#### Possible solution design or implementation
Substitute `0.10` for `0.13` in `Cargo.toml` and make the changes to the repository's code.

#### Additional context
The migration to [`piecrust`](https://github.com/dusk-network/rusk/issues/760) is using `plonk@0.13` first. As such the progress on this is being tracked there as well.

### PR DESCRIPTION
**Pull request by ureeves**
Migrates the circuits to use the improved PlonK API. This required some changes in both the `code-hasher` and in `rusk-recovery` for proper key generation.

Resolves #764

### CODE REVIEW

**Comment by ureeves** [comment]
Just like #762, this PR will fail to build in the CI, and that's why it is being merged into the `piecrust` branch. It is part of a larger effort to [migrate to `piecrust`](https://github.com/dusk-network/rusk/issues/760).
To ensure this builds and test correctly, please check the branch out and run the following list of commands:
```sh
source .env
make keys
cd circuits/transfer
cargo t --release --features builder
```

**Review by krl — APPROVED** [review]

**Review by OsDusk — APPROVED** [review]

**Review by herr-seppia — APPROVED** [review]
LGTM

**Review by miloszm — APPROVED** [review]
LGTM, I'd only remove superfluous `expect` statements in tests

File: circuits/transfer/tests/send_to_contract_obfuscated.rs [hunk 1]
```diff
@@ -74,48 +73,38 @@ fn create_random_circuit<R: RngCore + CryptoRng>(
...
 
-    let proof = circuit
-        .prove(&pp, &pk, TRANSCRIPT_LABEL)
-        .expect("Failed to prove circuit");
+    let circuit = create_random_circuit(rng, true);
 
-    let pi = circuit.public_inputs();
+    let (proof, public_inputs) = prover
+        .prove(rng, &circuit)
+        .expect("Proving the circuit should be successful");
```
[the remark this answers is not in the document]
Comment (ureeves) [reply circuits/transfer/tests/send_to_contract_obfuscated.rs:85]: I actually prefer this way. This way, we still get to see the error if it panics and get to add some context info.

**Pull request merged**